sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -940,14 +942,21 @@ public CosmosAsyncClient buildAsyncClient() {
         buildConnectionPolicy();
         CosmosAsyncClient cosmosAsyncClient = new CosmosAsyncClient(this);
         if (proactiveContainerInitConfig != null) {
+
+            cosmosAsyncClient.recordOpenConnectionsAndInitCachesStart(proactiveContainerInitConfig.getCosmosContainerIdentities());
+
             Duration aggressiveProactiveConnectionEstablishmentTimeWindow = proactiveContainerInitConfig
                     .getAggressiveProactiveConnectionEstablishmentDuration();
             if (aggressiveProactiveConnectionEstablishmentTimeWindow != null) {
                 cosmosAsyncClient.openConnectionsAndInitCaches(aggressiveProactiveConnectionEstablishmentTimeWindow);
             } else {
                 cosmosAsyncClient.openConnectionsAndInitCaches();
             }
+
```

Comment:
I like how you decoupled `ProactiveOpenConnectionsProcessor` from `warmUpDuration`.